package.json

```diff
@@ -21,7 +21,8 @@
     "bluebird": "^3.3.4",
     "htmlencode": "^0.0.4",
     "lodash": "^4.17.11",
-    "request": "^2.83.0"
+    "request": "^2.83.0",
+    "stringstream": "~0.0.5"
```

Comment:
I don't think we should add a direct dependency on this package, as it's not required for the library to work.

Applications using this library will have to bump the stringstream package in their own lockfiles to get the fixed version; bumping it in the lockfile in this repo only affects the version used when developing the library itself:

https://yarnpkg.com/blog/2016/11/24/lockfiles-for-all/

> When you publish a package that contains a `yarn.lock`, any user of that library will not be affected by it. When you install dependencies in your application or library, only your own `yarn.lock` file is respected. Lockfiles within your dependencies will be ignored.

I was surprised to discover that Yarn has no way to upgrade indirect dependencies (https://github.com/yarnpkg/yarn/issues/4986). Presumably that's what prompted you to add the direct dependency?

Given all that: I think leaving `package.json` as is and running `yarn upgrade request`, or even `yarn upgrade`, would be a better was to resolve https://github.com/intercom/intercom/issues/143632.

Reply:
Yeah I dislike this too, but it seems to be the default behaviour of yarn in this case. I could just not commit the package json change, and the yarn lock would resolve to 0.0.6 in the future?

Reply:
Update: I [amended](https://github.com/intercom/intercom-node/compare/b65cfe2c6fa8ed1c7c4139b4a38dc14edba3fd48..ba19772ee1496bdb27c13806c0258c1adafe783f) this PR so that it only ships the updated `yarn.lock`. I think it cleanly addresses the bump for the vulnerability without adding additional explicit dependencies as it did before.

Reply:
Looks like that's what the autoupdate on another repo did: https://github.com/intercom/ember-computed-template-string/pull/27/files

Reply:
👍 we should probably turn on Dependabot in this repo too!

Reply:
Good shout, done!